core/engine/src/value/mod.rs

```diff
@@ -292,48 +394,39 @@ impl JsValue {
     #[must_use]
     #[allow(clippy::float_cmp)]
     pub fn is_integral_number(&self) -> bool {
```

Comment:
Not blocking this PR, but I think this function name is probably a bit missleading and we should not have it at all. Same goes for `is_integer`. The spec defines these names as mathematical terms. As far as I understand the spec definition, there is no restriction in max values, so our function names do not match with the spec definition, since we depend on i32.

Both functions are not used at all (only used once by temporal), So I would be in favor of removing them in favor of the new `as_integer32` (@hansl any reason to not name it `as_i32`?).

Reply:
`as_i32()` would probably be good enough. People can match on the output.

There's a lot of weirdness around integers and I'm not sure if it should be within this PR scope. I'm thinking that `Integer32` should really be `Integer64`, since it should represent anything that _can_ be an integer in JS. Not sure why 32-bits was selected in the previous APIs, but didn't want to make that change.